src/Runner.Worker/ActionManager.cs

```diff
@@ -613,14 +613,35 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 }
 #endif
 
-                // repository archive from github always contains a nested folder
+                // repository archive from github always contains a nested folder, the nested folder contains the short SHA for the ref
+                // ex: actions/checkout@master the nested folder looks like actions-checkout-01aeccc
+                // However, when you reference action using tag, the SHA in the folder name is not really the commit SHA, it's the SHA for the tag itself.
```

Comment:
I didn't find an ideal solution for handle reference actions by tag, since github api only supports `GET: /repos/org/repo/tags`, it doesn't allow query by tagname, so i have to make multiple api calls to get the tag object then find out which commit the tag is pointing to.

Reply:
https://api.github.com/repos/Microsoft/terminal/git/ref/tags/1810.02002

https://api.github.com/repos/:org/:repo/git/ref/tags/:tagname (can include `/`) -- https://developer.github.com/v3/git/refs/#get-a-single-reference